### ISSUE TITLE
Operands of '\" must be numberic of SMDegrain in line 175, 446

### ISSUE THREAD

**Issue by kedaitinh12** [Dogway/Avisynth-Scripts#17]
I meet error when use this script:

> sdwcomp=DotKills(iterations=4).Rainbow_Smooth2(5,200)
smdc=SMDegrain(1,1000,1000,true, mfilter=sdwcomp,subpixel=3,amp=2,prefilter=sdwcomp.expr("x 0.5 * 0.125 x x[0,-1] - abs 32 scaleb > x x[0,-1] > x 32 scaleb + x 32 scaleb - ? x[0,-1] ? * + 0.125 x x[0,1] - abs 32 scaleb > x x[0,1] > x 32 scaleb + x 32 scaleb - ? x[0,1] ? * + 0.125 x x[-2,0] - abs 32 scaleb > x x[-2,0] > x 32 scaleb + x 32 scaleb - ? x[-2,0] ? * + 0.125 x x[2,0] - abs 32 scaleb > x x[2,0] > x 32 scaleb + x 32 scaleb - ? x[2,0] ? * +").sGConvolution("-1 2 -1 2 0 2 -1 2 -1",chroma=true)) #expr line made as imitation to https://web.archive.org/web/20150602063631/http://home.earthlink.net/~tacosalad/video/dotcrawl.html and sGConvolution from https://forum.doom9.org/showpost.php?p=1751654&postcount=5
mt_merge(last,smdc,scombmask,u=3,v=3)
mt_merge(last,smdc,scombmask,u=3,v=3)

But when change to NotSMDegrain() of real.finder, I don't meet this error

**Comment by Dogway** [comment]
Yes, it's a problem with sh_GetUserGlobalIMTint(), it outputs blank, by setting it to sh_GetUserGlobalIMTint(true) it works back again.
On the linked algo, I worked on it yesterday but found it subpar compared to wonkey_monkey algo:
```
Expr("x[0,1] x[0,0] J@ 32 scaleb - O@ J 32 scaleb + P@ clip B^ x[-1,0] O P clip D^ x[1,0] O P clip E^ x[0,-1] O P clip G^
       B D E G + + + 0.125 * J 0.5 * +","")
```
`Expr("x[0,1] 2 * x[-1,0] 2 * x[1,0] 2 * x[0,-1] 2 * + + + x[-1,-1] - x[1,-1] - x[-1,1] - x[1,1] - 0.250 *","")`

**Comment by Dogway** [comment]
Fixed: 1c3ef4f642d71ce9f957192e2a9e9d32e9874a3c

**Comment by kedaitinh12** [comment]
Thanks
